From 6b3d887c4579db754fa30893f1e51a437f67f932 Mon Sep 17 00:00:00 2001
From: Mythri P K <mythri.p.k@intel.com>
Date: Tue, 20 May 2014 20:03:51 +0530
Subject: [PATCH 15/39] ASoC: soc-fw: Add num_controls to fw_dapm_widget

Issue: GMINL-3679

use num_controls instead of kcontrol count, As kcontrol
is not used for anything else remove it.

Change-Id: I9aec0cc459176f02246b00f1823b78fc393a71a7
Signed-off-by: Mythri P K <mythri.p.k@intel.com>
---
 include/uapi/sound/asoc.h | 8 +++++---
 sound/soc/soc-fw.c        | 6 +++---
 2 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/include/uapi/sound/asoc.h b/include/uapi/sound/asoc.h
index 89ec6ec..aa777f0 100644
--- a/include/uapi/sound/asoc.h
+++ b/include/uapi/sound/asoc.h
@@ -892,9 +892,11 @@ struct snd_soc_fw_dapm_widget {
 	__u16 event_flags;
 	__u16 event_type;
 
-	/* kcontrols that relate to this widget */
-	struct snd_soc_fw_kcontrol kcontrol;
-	/* controls follow here */
+	__u16 num_kcontrols;
+	/*
+	 * kcontrols that relate to this widget
+	 * follow here
+	 */
 } __attribute__((packed));
 
 /*
diff --git a/sound/soc/soc-fw.c b/sound/soc/soc-fw.c
index 2f610d9..6ff5f0d 100644
--- a/sound/soc/soc-fw.c
+++ b/sound/soc/soc-fw.c
@@ -1121,14 +1121,14 @@ static int soc_fw_dapm_widget_create(struct soc_fw *sfw,
 	widget.index = sfw->index;
 
 	sfw->pos += sizeof(struct snd_soc_fw_dapm_widget);
-	if (w->kcontrol.count == 0) {
+	if (w->num_kcontrols == 0) {
 		widget.num_kcontrols = 0;
 		goto widget;
 	}
 
 	control_hdr = (struct snd_soc_fw_control_hdr *)sfw->pos;
 	dev_dbg(sfw->dev, "ASoC: widget %s has %d controls of type %x\n",
-		w->name, w->kcontrol.count, control_hdr->index);
+		w->name, w->num_kcontrols, control_hdr->index);
 
 	switch (SOC_CONTROL_GET_ID_INFO(control_hdr->index)) {
 	case SOC_CONTROL_TYPE_VOLSW:
@@ -1140,7 +1140,7 @@ static int soc_fw_dapm_widget_create(struct soc_fw *sfw,
 	case SOC_CONTROL_TYPE_BOOL_EXT:
 	case SOC_CONTROL_TYPE_RANGE:
 	case SOC_DAPM_TYPE_VOLSW:
-		widget.num_kcontrols = w->kcontrol.count;
+		widget.num_kcontrols = w->num_kcontrols;
 		widget.kcontrol_news = soc_fw_dapm_widget_dmixer_create(sfw,
 			widget.num_kcontrols);
 		if (!widget.kcontrol_news) {
-- 
1.9.1

